polaris/dataset/_dataset.py

```diff
@@ -432,10 +495,13 @@ def cache(self, cache_dir: Optional[str] = None) -> str:
 
         self.to_json(self.cache_dir)
 
-        if self.zarr_root_path is not None:
-            self.zarr_root_path = fs.join(self.cache_dir, "data.zarr")
+        if self.uses_zarr:
+            self.zarr_root_path = dmfs.join(self.cache_dir, "data.zarr")
             self._zarr_root = None
 
+        if verify_checksum and self._md5sum is not None:
+            self.verify_checksum(md5sum=self._md5sum)
```

Comment:
**Question**
This feels a little weird. Wouldn't this do the same thing?
```suggestion
        if verify_checksum:
            self.verify_checksum()
```

Reply:
It's not exactly the same thing, because it would throw an exception if `self._md5sum` would be `None`. On second thought, however, I think that is better behavior? If you specify `verify_checksum`, it should fail if there is no checksum to verify.